stdlib/polylinesimp/polylinesimp.flux

```diff
@@ -0,0 +1,65 @@
+// Package polylinesimp provides methods for polyline simplication which is an efficient way of downsampling curves retaining the moments of variation throughout the path.
+// This class of algorithms enable efficient rendering of graphs and visualizations without having to load the entire corpus of data points into memory.
+// This is done by reducing the number of vertices that donot contribute significantly to the convexity and concavity of the shape.
+//
+// ## Metadata
+// introduced: 0.7.0
+//
+package polylinesimp
+
+
+// rdp (Ramer Douglas Peucker algorithm) is a polyline simplification function that decimates a curve composed of line segments into a downsampled yet visually indistinguishable curve with fewer points.
+//
+// ## Parameters
+// - valColumn: column that corresponds to the Y axis of the given curve. (optional) (default valColumn: _value)
+// - timeColumn: column that corresponds to the X axis of the given curve. (optional) (default timeColumn: _time)
+// - epsilon: user defined maximum tolerance value that determines the amount of compression. (optional) (epsilon should be greater than 0.0)
+// - retention: user defined retention rate, which specifies the percentage of points to be retained after downsampling. (optional) (Retention rate should be between 0.0 and 100.0)
```

Comment:
What happens if both epsilon and retention are specified? 

What is the default if neither is specified?

Reply:
@nathanielc If both epsilon and retention are provided we exit with an error message **"Please provide either epsilon or retention rate and not both"**

if both epsilon and retention are not specified the algorithm will run the automatic case and try to make an educated guess for the epsilon value.